notebooks/228-clip-zero-shot-image-classification/228-clip-zero-shot-image-classification.ipynb

```diff
@@ -0,0 +1,281 @@
+    "probs = logits_per_image.softmax(dim=1).detach().numpy()  # we can take the softmax to get the label probabilities\n",
+    "visualize_result(image, input_labels, probs[0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Convert model to OpenVINO Intermediate Representation (IR) format.\n",
+    "\n",
+    "For best results with OpenVINO we recommend converting the model to OpenVINO IR format. OpenVINO supports PyTorch via ONNX conversion. We will use `transformers.onnx` for exporting the ONNX model from HuggingFace transformers library. We need to provide Model ID on huggingface.co or path on disk to load model from and directory for saving ONNX model. Optionally, additional parameters like  `--feature` - type of model output and `--opset` target ONNX opset for export can be provided, but in our case default parameters should be enough. More information about this functionality and supported models can be found in [HuggingFace documentation](https://huggingface.co/docs/transformers/serialization). We will use `mo.convert_model` functionality to convert the ONNX model. The `mo.convert_model` Python function returns an OpenVINO model ready to load on device and start making predictions. We can save it on disk for next usage with `openvino.runtime.serialize`.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python -m transformers.onnx --model openai/clip-vit-base-patch16 clip-vit-base-patch16"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from openvino.runtime import serialize\n",
+    "from openvino.tools import mo\n",
+    "\n",
+    "ov_model = mo.convert_model('clip-vit-base-patch16/model.onnx', compress_to_fp16=True)\n",
+    "serialize(ov_model, 'clip-vit-base-patch16/model.xml')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run OpenVINO model\n",
+    "\n",
+    "The steps for making predictions with OpenVINO CLIP model are the same like for PyTorch model. Let's check model result."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from scipy.special import softmax\n",
+    "from openvino.runtime import Core\n",
+    "\n",
+    "# create OpenVINO core object instance\n",
+    "core = Core()\n",
+    "# compile model for loading on device\n",
+    "compiled_model = core.compile_model(ov_model)\n",
+    "# obtain output tensor for getting predictions\n",
+    "logits_per_image_out = compiled_model.output(0)\n",
+    "# run inference on preprocessed data and get image-text similarity score\n",
+    "ov_logits_per_image = compiled_model(dict(inputs))[logits_per_image_out]\n",
```

Comment:
inputs are the same as for PyTorch path? I assume this means CLIPProcessor output or something similar?

Reply:
yes, they are the same. The only difference is conversion custom python dict-like object represented batch to dictionary.
Do you mean I need to add discription to highlight it?